zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -291,7 +314,8 @@ public Interpreter getCurrentRepl() {
 
     String replName = getRequiredReplName(buffer);
     if (replName != null && cursor > replName.length()) {
-      cursor -= replName.length() + 1;
+      String body = buffer.substring(replName.length() + 1);
+      cursor -= replName.length() + 1 + body.indexOf(body.trim());
```

Comment:
I think maybe the trim() on body should go before the substring? say for example
```
  %foo    select


* from table
```

if there's whitespace before replName